### ISSUE TITLE
Implicit/default merge behaves different than explicit merge

### ISSUE THREAD

**Issue by HeavyWombat** [geofffranks/spruce#267]
Working with some Prometheus YAML files, we found that Spruce behaves quite different if you use `(( merge ))` explicitly. It works fine with the implicit default merge, but fails if you explicitly define `(( merge ))` or `(( merge on key ))`.

Example that shows the basic setup in which default identifier key names (`name`) are used: https://play.spruce.cf/#43d9053495d1732fd4ac34ca6417a4ce

Same example with non-standard identifier key names (`job-name`), which therefore require an explicit `merge on key`: https://play.spruce.cf/#a6a6620555a56ed417bbd0ec583f6384

Just to show it in this issue, the raw setup here as well:
```
serverFiles:
  prometheus.yml:
    scrape_configs:
    - job_name: one
```

```
serverFiles:
  prometheus.yml:
    scrape_configs:
    - (( merge on job_name ))
    - job_name: two
    - (( delete job_name "one" ))
```

Switching the order or `merge` and `delete` does not work, too. In fact, I believe this hints to even another bug in Spruce.

**Comment by jhunt** [comment]
I believe this is by design.

**Comment by geofffranks** [comment]
Fix released in v1.18.2
